django-stubs/core/cache/backends/redis.pyi

```diff
@@ -0,0 +1,54 @@
+from collections.abc import Mapping
+from datetime import timedelta
+from typing import Any, Callable, Iterable, Protocol, SupportsInt, overload, type_check_only
+
+from _typeshed import ReadableBuffer, Unused
+from django.core.cache.backends.base import BaseCache
+from redis.client import Redis
+from redis.connection import BaseParser, ConnectionPool
```

Comment:
I think handling ImportError should work fine:

```suggestion
try:
    from redis.client import Redis
    from redis.connection import BaseParser, ConnectionPool
except ImportError:
    Redis: TypeAlias = Any
    BaseParser: TypeAlias = Any
    ConnectionPool: TypeAlias = Any
```

Not sure if `Any` or `_typeshed.Incomplete` is preferred. I would be fine with either one.

Reply:
Hum, I still get `Library stubs not installed for "redis.client"  [import-untyped]`. Maybe we should install the redis types, similar to the ones for pyyaml?

Reply:
> Maybe we should install the redis types, similar to the ones for pyyaml?

Interesting. It didn't occur to me that `types-pyyaml` is an unconditional dependency of `django-stubs`, but you are right.

On the other hand: I've seen lots of Django projects over my career and exactly 0 of them used either Redis or Django's PyYAML serialization. :stuck_out_tongue: So it seems a bit of a waste to have these as unconditional dependency, when these dependencies are optional in Django itself.

I think adding it as an optional dependency, and updating `requirements.txt` with `-e .[compatible-mypy,redis]`, should solve this.